R/SS_write.R

```diff
@@ -0,0 +1,110 @@
+#' Write all Stock Synthesis input files for a model
+#'
+#' Writes all the input files for a Stock Synthesis model using the list
+#' created by [SS_read()]
+#' (presumably after modification of one or more elements)
+#' using the `SS_write*()` functions for the four or
+#' five model input files.
+#'
+#' @param inputs list created by [SS_read()]
+#' @template dir
+#' @param files vector of file types to write. The default is the four
+#' standard input files plus wtatage.ss if used. The ss.par file is not
+#' yet supported by this function or [SS_read()].
+#' @template overwrite
+#' @template verbose
+#'
+#' @author Ian G. Taylor
+#'
+#' @export
+#' @seealso
+#' * [SS_read()] creates the list that is used by this function.
+#' * [SS_writestarter()], [SS_writedat()], [SS_writectl()],
+#' [SS_writeforecast()], and [SS_writewtatage()] are used to write the
+#' input files.
+
+#' @examples
+#' \dontrun{
+#' # read inputs to modify the data file
+#' inputs <- SS_read(
+#'   dir = system.file("extdata", "simple_3.30.13", package = "r4ss")
+#' )
+#'
+#' # modify the starter file (use the par file)
+#' inputs$start$init_values_src <- 1
+#'
+#' # modify the data file (remove age comps from years prior to 1990)
+#' inputs$dat$agecomp <- inputs$dat$agecomp %>% dplyr::filter(Yr >= 1990)
+#'
+#' # modify the control file (turn off early recdevs and change range of yrs)
+#' inputs$ctl$recdev_early_phase <- -abs(inputs$ctl$recdev_early_phase)
+#' inputs$ctl$MainRdevYrFirst <- 1980
+#'
+#' # write the files to a new folder within the source directory
+#' SS_write(inputs = inputs,
+#'          dir = file.path(inputs$dir, "modified_inputs"))
+#' }
+
+SS_write <- function(inputs,
+                     dir = NULL,
+                     files = c("dat", "ctl", "start", "fore", "wtatage"),
```

Comment:
Seems like you could get rid of this argument and just write the files that are in the list. This would force users to always write the starter file though b/c you could never remove that element from the list. Anyway, I think that the input argument `files` makes things overly complicated and if users want to write just a few rather than all they should just use the individual functions.

Reply:
The workflow that I used for the `write_inputs_ling()` function which preceded this one was to read all inputs files at the start, then modify a subset and only write out the ones that got modified (e.g. for a particular sensitivity could be either control or data: https://github.com/iantaylor-NOAA/Lingcod_2021/blob/main/R/run_sensitivities.R). 

The arguments in the SS_write*() functions depend on the file being written ("outfile" is used for data and control files, while "dir" is used for starter and forecast since the names are fixed). So for me it requires less thought to just run this wrapper function each time.

I did simplify from `write_inputs_ling()` by removing control of the filenames under the assumption that the user would just modify the values in `inputlist$start`.

Reply:
Issue resolved via discussion and revisions to `SS_write()` in 2862de3. `SS_write()` will now write whatever elements are present in the list. It's not more efficient (may overwrite existing file with no changes), but it's simpler and easier to understand.